ocs_ci/ocs/rados_utils.py

```
@@ -538,12 +538,29 @@ def corrupt_pg(osd_deployment, pool_name, pool_object):
+        f'"runAsUser": 0}}, "volumeMounts": [{{"mountPath": "/var/lib/ceph/osd/ceph-{osd_id}", '
+        f'"name": "{bridge_name}", "subPath": "ceph-{osd_id}"}}]}}}}]'
     )
     osd_deployment.ocp.patch(
         resource_name=osd_deployment.name, params=patch_change, format_type="json"
     )
+
+    # Wait for the OSD pod to restart with corrupted data before issuing
+    # deep-scrub. If deep-scrub runs while the corrupted OSD is down, it only
+    # checks healthy replicas and finds no inconsistency.
+    logger.info(
+        f"Waiting for OSD pod {osd_deployment.name} to restart after corruption"
```

Ruff (0.15.21)

[warning] Logging statement uses f-string

(G004)